### ISSUE TITLE
Create run alert only after last retry fails

### ISSUE THREAD

**Issue by robhudson** [mozilla/telemetry-analysis-service#496]
Sentry:
https://sentry.prod.mozaws.net/operations/atmo-stage/issues/389740/events/10824516/

Currently an alert record gets created after the first failure. Upon retry and failure another alert tries to get created but fails due to a primary key constraint.

I think it would be preferable if we only created the alert after the last retry and failure so the user gets one email after all attempts have failed. Which is better than getting an email and the 2nd retry worked.

**Comment by maurodoglio** [comment]
An alternative would be to send an email on every failure and show in the content how many runs were attempted so far and how many are left, e.g.
> Try 2 out of 3

This is what Airflow does and I find it very useful. Also, Airflow indicates in the email subject if the job is app for retry or if it's in a final failed state.

**Comment by robhudson** [comment]
I like the idea of sending an email for each try and annotating it.

The backend doesn't seem to support that and I'm curious why the alert records aren't cleared once an email is sent. But changing the one-to-one mapping to a many-to-one would allow this, so multiple alerts could be assigned to a run.

**Comment by jezdez** [comment]
This seems like an actual big regression since it prevents jobs from retrying successfully.

**Comment by jezdez** [comment]
@robhudson The alert records were supposed to stay around since it would allow seeing which run would generate a run. The relationship is a one-to-one mapping since that's the level we cared about at the time when we didn't have more elaborate retrying.

Since this is now a blocker to the core functionality of ATMO I don't agree with @maurodoglio's proposals as that is a big change in functionality. Let's open a separate ticket to develop that but first fix the immediate issue.

**Comment by jezdez** [comment]
https://sentry.prod.mozaws.net/operations/atmo-prod/issues/395279/

**Comment by jezdez** [comment]
To elaborate a bit more, this has become critical since the change in 58baeb58ce9d49347ee1f65d01ac7424380d2c69 moved the retrying from Celery's retry mechanism into one that acts in-process, effectively making any exception that is raised during the retry that is not a Boto connection error basically a stopper for the whole run. I'm contemplating reverting 58baeb58ce9d49347ee1f65d01ac7424380d2c69 because of that.
